src/components/CTreeview/CTreeview.js

```diff
@@ -252,15 +253,39 @@ export default {
       searchTree(this.items);
       return sortBy(leafs);
     },
+    getItemById(id) {
+      let item = null;
+      const searchTree = (items) => {
+        each(items, (child) => {
+          if (child[this.itemValue] === id) item = child;
+          if (child[this.itemChildren] && child[this.itemChildren].length > 0) {
+            searchTree(child[this.itemChildren]);
+          }
+        });
+      };
+      searchTree(this.items);
+      return item;
+    },
     loadData() {
       this.loadConnectorData().then((result) => {
         this.items = result.items || [];
         this.sendToEventBus('DataSourceChanged', this.dataSource);
       });
     },
     async getChildren(item) {
-      this.sendToEventBus('GetAsyncChildren', item);
-      // addItem();
+      const connector = this.options.connector;
+      const id = this.dataSource.connector.id;
+      const type = this.options.connectors[id];
+
+      if (connector) {
+        connector.getSourceData(type, this.dataSource).then((result) => {
+          if (result && item) {
+            // Can't add directly to item
+            // this.addItem(result, item[this.itemValue]);
+          }
+        });
+      }
+      // this.sendToEventBus('GetAsyncChildren', item);
```

Comment:
You can add this as `else` statement if connector does not exist and on `getSourceData` catch statement.